### PR TITLE
Use default templates as embedded resource 

### PR DESCRIPTION
**Pull request by moria97**
## Description
Change default templates as embedded resource instead of content files, which embed the templates inside the dll and increases the flexibility to reference nuget packages.

## Related issues

Addresses [issue #].

## Testing

Run existing tests on template collection provider.

### CODE REVIEW

**Review by sowu880 — APPROVED** [review]

**Comment by ACMoretxj** [comment]
@sowu880 Currently there are several tests failed in TemplateManagement subproject, it this in our expectations(like the tests are still under construction)?

File: src/Microsoft.Health.Fhir.TemplateManagement/Models/TemplateLayer.cs [hunk 1]
```diff
@@ -17,21 +18,27 @@ public class TemplateLayer : OCIArtifactLayer
     {
         public Dictionary<string, Template> TemplateContent { get; set; }
 
-        public static TemplateLayer ReadFromFile(string filePath)
+        public static TemplateLayer ReadFromEmbeddedResource()
```
Comment (qiwjin) [inline src/Microsoft.Health.Fhir.TemplateManagement/Models/TemplateLayer.cs:21]: i feel we can keep the original api and also add a new one ReadFromStream().
both ReadFromFile and ReadFromEmbeddedResource will call ReadFromStream.

File: src/Microsoft.Health.Fhir.TemplateManagement/Microsoft.Health.Fhir.TemplateManagement.csproj [hunk 2]
```diff
@@ -30,5 +30,15 @@
...
   </ItemGroup>
+  
+  <ItemGroup>
+    <EmbeddedResource Include="../../bin/DefaultTemplates.tar.gz">
+      <Link>DefaultTemplates.tar.gz</Link>
+    </EmbeddedResource>
+  </ItemGroup>
+
+  <Target Name="PreBuild" BeforeTargets="PreBuildEvent">
+    <Exec Command="mkdir ..\..\bin &amp; tar -zcvf ../../bin/DefaultTemplates.tar.gz -C ../../data/Templates/Hl7v2 ." />
```
Comment (qiwjin) [inline src/Microsoft.Health.Fhir.TemplateManagement/Microsoft.Health.Fhir.TemplateManagement.csproj:41]: nice, can you help to change the name into 'HL7v2DefaultTemplates'?

File: src/Microsoft.Health.Fhir.TemplateManagement/Models/TemplateLayer.cs (hunk 1 above)
Comment (moria97) [reply src/Microsoft.Health.Fhir.TemplateManagement/Models/TemplateLayer.cs:21]: Sure. Have added the ReadFromFile() api back.

File: src/Microsoft.Health.Fhir.TemplateManagement/Microsoft.Health.Fhir.TemplateManagement.csproj (hunk 2 above)
Comment (moria97) [reply src/Microsoft.Health.Fhir.TemplateManagement/Microsoft.Health.Fhir.TemplateManagement.csproj:41]: Maybe it's better to add a new "Renaming" PR as this name is referenced in multiple places @sowu880

File: src/Microsoft.Health.Fhir.TemplateManagement/Utilities/StreamUtility.cs [hunk 3]
```diff
@@ -74,5 +74,16 @@ public static string CalculateDigestFromSha256(byte[] content)
...
+
+        public static string CalculateDigestFromSha256(Stream stream)
+        {
+            using SHA256 mySHA256 = SHA256.Create();
+            string hashedValue = "sha256:";
+            byte[] hashData = mySHA256.ComputeHash(stream);
+            string[] hashedStrings = hashData.Select(x => string.Format("{0,2:x2}", x)).ToArray();
+            hashedValue += string.Join(string.Empty, hashedStrings);
+            return hashedValue;
+        }
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.TemplateManagement/Utilities/StreamUtility.cs:86]: It seems these two functions share most logics. Could we convert the input stream to byte[] and call the original one?

Comment (moria97) [reply src/Microsoft.Health.Fhir.TemplateManagement/Utilities/StreamUtility.cs:86]: Yeah, they are sharing same logic in get the hash value.  Converting to byte array will introduce extra cost than computing from a stream directly. I feel it's fine to keep two methods here.

**Review by BoyaWu10 — APPROVED** [review]

**Review by ACMoretxj — APPROVED** [review]

**Pull request merged**